### PR TITLE
Move End-links or Footer References to in-line links

### PR DESCRIPTION
**Pull request by kingthorin**
- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- End-links and footer references changed to in-line links.
- A very few minor corrections.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_WSTG-CRYPST-001.md [hunk 1]
```diff
@@ -549,7 +553,7 @@ SCAN COMPLETED IN 9.68 S
 
 #### Example 6. Testing SSL/TLS with testssl.sh
 
-Testssl.sh [[38]] is a Linux shell script which provides clear output to facilitate good decision making. It can not only check web servers but also services on other ports, supports STARTTLS, SNI, SPDY and does a few check on the HTTP header as well.
+[Testssl.sh](https://testssl.sh/) is a Linux shell script which provides clear output to facilitate good decision making. It can not only check web servers but also services on other ports, supports STARTTLS, SNI, SPDY and does a few check on the HTTP header as well.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_WSTG-CRYPST-001.md]: https://github.com/drwetter/testssl.sh
What do you think about this?

**Comment by ThunderSon** [comment]
It's hilarious how big the first document is. 1.2k lines ...

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_WSTG-CRYPST-001.md [hunk 2]
```diff
@@ -4,13 +4,13 @@
 
 Sensitive data must be protected when it is transmitted through the network. Such data can include user credentials and credit cards. As a rule of thumb, if data must be protected when it is stored, it must be protected also during transmission.
 
-HTTP is a clear-text protocol and it is normally secured via an SSL/TLS tunnel, resulting in HTTPS traffic [[1]]. The use of this protocol ensures not only confidentiality, but also authentication. Servers are authenticated using digital certificates and it is also possible to use client certificate for mutual authentication.
+HTTP is a clear-text protocol and it is normally secured via an SSL/TLS tunnel, resulting in [HTTPS traffic](https://www.ietf.org/rfc/rfc5246.txt). The use of this protocol ensures not only confidentiality, but also authentication. Servers are authenticated using digital certificates and it is also possible to use client certificate for mutual authentication.
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_WSTG-CRYPST-001.md]: Sure

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_WSTG-CRYPST-001.md (hunk 1 above)
Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_WSTG-CRYPST-001.md]: Sure

**Comment by kingthorin** [comment]
Done

**Pull request merged**